src/Helper.php

```diff
@@ -109,15 +109,47 @@ public static function get_shipment_carriers() {
 
     /**
      * Returns the available aliases.
+     * @param bool $used
      *
-     * @since 0.1.0
+     * s@since 0.1.0
      * @return array List of aliases.
      */
-    public static function get_aliases() {
-        return [
-            'delivered' => __( 'Delivered', 'trackmage' ),
-            'shipped'   => __( 'Shipped', 'trackamge' ),
-        ];
+    public static function get_aliases($used = false) {
+        $workspaceId = get_option( 'trackmage_workspace', 0 );
+        $cachedAliases = get_transient('trackmage_order_statuses');
+        $aliases = [];
+        if($cachedAliases && isset($cachedAliases[$workspaceId])) {
+            $aliases = $cachedAliases[$workspaceId];
+        }elseif($workspaceId !== 0){
+            try {
+                $client = Plugin::get_client();
+                $guzzleClient = $client->getGuzzleClient();
+                $response = $guzzleClient->get("/workspaces/{$workspaceId}/order_statuses");
+                $content = $response->getBody()->getContents();
+                $data = json_decode($content, true);
+
+                foreach ($data['hydra:member'] as $status) {
+                    $name = $status['name'];
+                    $aliases[$name] = __(ucfirst($name),'trackmage');
+                }
+
+                $cachedAliases[$workspaceId] = $aliases;
+                set_transient('trackmage_order_statuses', $cachedAliases, 3600);
+            } catch( ApiException $e ) {
+
+            }
+
+        }
+        if($used){
```

Comment:
`bool $used` is a clear sign we need to move this logic into a separate function `get_used_aliases()`

Reply:
@karser okay, I'll create separate function.